test/Components/ImageTest.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Test\Components;
+
+use Gui\Application;
+use Gui\Components\Image;
+use PHPUnit\Framework\TestCase;
+
+class ImageTest extends TestCase
+{
+    public function testMissingFile()
+    {
+        $this->setExpectedException(
+            \Exception::class,
+            'The file could not be found.'
+        );
+        
+        $image = new Image(array(), null, new Application());
```

Comment:
can you use [] instead of array()

Reply:
Sorry, originally I did but I wasn't sure if PHP 5 was compatible, I can do that for you. Can I also use type hints for arguments and return values?